### ISSUE TITLE
Can't use multiple values with IN operator?

### ISSUE THREAD

**Issue by reboxer** [mariadb-corporation/mariadb-connector-nodejs#117]
Today i tried to do a basic query to get some things from the database using a simple select query with the IN operator and a placeholder 
```js
await pool.query("SELECT * FROM table WHERE id IN (?)", ["2, 3"])
```
but only the data for id 2 is being returned in the response instead of both, but it works when using a template string and using `Array.join()` to put the values together
```js
await pool.query(`SELECT * FROM table WHERE id IN (${[2, 3].join(", ")})`)
```

should't the first one work? or am i doing something wrong?

**Comment by rusher** [comment]
This is indeed one of the rare differences compare to [mysql ](https://www.npmjs.com/package/mysql).

At some point, Connector will use a protocol that will permit to send SQL and parameters separately without any need to parsing query and escaping parameters. This will add some new possibilities like caching and avoiding metadata, ... 
That will probably to gain 20% more performance and this can be done only if keeping API this way (permitting [PREPARE](https://mariadb.com/kb/en/prepare-statement/)).

**Comment by pochwerbs** [comment]
I don't know if this is related to OP's question, but I'm having troubles using IN statement.

```javascript
  await client.query(
    `SELECT count(1) as total FROM store_product WHERE url IN(?) AND status = ?;`, 
    [['url', 'another url', 'an extra one'], 1],
  )
```

The first placeholder should bind to the array `['url', 'another url', 'an extra one']` and the second placeholder should bind to number `1`.

I get the following error:
```
  code:"ER_OPERAND_COLUMNS"
  errno:1241
  fatal:false
  message:"(conn=61417, no: 1241, SQLState: 21000) Operand should contain 1 column(s)
  sql: SELECT count(1) as total FROM store_product WHERE url IN(?) AND status = ?; - parameters:[url,another url,an extra one,1]"
  sqlState:"21000"
```

Is there a way to pass an array as a placeholder value?

**Comment by rusher** [comment]
In fact, my first answer is incomplete Array parameters are actually supported, but it differs compare to mysql2 and mysql. 

for mysql2 and mysql, Arrays are turned into list, e.g. ['a', 'b'] turns into 'a', 'b'
mariadb handle array parameter with parenthesis , so ['a', 'b'] turns into ('a', 'b') 

I don't see any possibility of injection removing that parenthesis, but need to check. This would concern [connection.escape](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/documentation/promise-api.md#connectionescapevalue--string) too

Then this would work:
```
await pool.query("SELECT * FROM table WHERE id IN (?)", [[2, 3]])
```

**Comment by rusher** [comment]
Closing this issue, since resolved since 4 years (version 2.1.0). 
Example : 
```
    await conn.query('CREATE TABLE arrayParam (id int, val varchar(10))');
    await conn.query("INSERT INTO arrayParam VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd')");
    const rows = await conn.query('SELECT * FROM arrayParam WHERE val IN (?)', [['b', 'c', 1]]);
```
Last command will be escaped as :
```
SELECT * FROM arrayParam WHERE val IN ('b','c',1) 
```
